File: backend/app/api/routes/applications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.dependencies import get_current_user
from app.database.models.user import User
from app.database.session import get_db
from app.schemas.community_application import CommunityApplicationResponse
from app.services.community_application_service import (
    CommunityApplicationService,
)

router = APIRouter()

community_application_service = CommunityApplicationService()


@router.get(
    "/me",
    response_model=list[CommunityApplicationResponse],
)
def list_my_applications(
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return community_application_service.list_my_applications(
        db=db,
        current_user=current_user,
    )


@router.post(
    "/{application_id}/accept",
    response_model=CommunityApplicationResponse,
)
def accept_application(
    application_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return community_application_service.accept_application(
        db=db,
        current_user=current_user,
        application_id=application_id,
    )


@router.post(
    "/{application_id}/reject",
    response_model=CommunityApplicationResponse,
)
def reject_application(
    application_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return community_application_service.reject_application(
        db=db,
        current_user=current_user,
        application_id=application_id,
    )


@router.post(
    "/{application_id}/cancel",
    response_model=CommunityApplicationResponse,
)
def cancel_application(
    application_id: int,
    current_user: User = Depends(get_current_user),
    db: Session = Depends(get_db),
):
    return community_application_service.cancel_application(
        db=db,
        current_user=current_user,
        application_id=application_id,
    )

File: frontend/src/services/applications.ts
import { api } from "./api";
import type {
  CommunityApplication,
  CommunityApplicationCreate,
} from "@/types/application";

export async function createCommunityApplication(
  communityId: number | string,
  payload: CommunityApplicationCreate
) {
  const { data } = await api.post<CommunityApplication>(
    `/communities/${communityId}/applications`,
    payload
  );

  return data;
}

export async function getCommunityApplications(
  communityId: number | string
) {
  const { data } = await api.get<CommunityApplication[]>(
    `/communities/${communityId}/applications`
  );

  return data;
}

export async function getMyApplications() {
  const { data } = await api.get<CommunityApplication[]>(
    "/applications/me"
  );

  return data;
}

export async function acceptApplication(applicationId: number) {
  const { data } = await api.post<CommunityApplication>(
    `/applications/${applicationId}/accept`
  );

  return data;
}

export async function rejectApplication(applicationId: number) {
  const { data } = await api.post<CommunityApplication>(
    `/applications/${applicationId}/reject`
  );

  return data;
}

export async function cancelApplication(applicationId: number) {
  const { data } = await api.post<CommunityApplication>(
    `/applications/${applicationId}/cancel`
  );

  return data;
}
